src/pages/iou/IOUCurrencySelection.js

```diff
@@ -0,0 +1,239 @@
+                />
+                <View style={[styles.flex1, styles.w100]}>
+                    <View style={[styles.flex1]}>
+                        <View style={[styles.ph5, styles.pv3]}>
+                            <TextInputWithFocusStyles
+                                styleFocusIn={[styles.textInputReversedFocus]}
+                                ref={el => this.textInput = el}
+                                style={[styles.textInput]}
+                                value={this.state.searchValue}
+                                onChangeText={(searchValue = '') => {
+                                    const {currencyOptions} = getCurrencyListForSections(
+                                        this.getCurrencyOptions(this.props.currencyList),
+                                        searchValue,
+                                    );
+                                    this.setState({
+                                        searchValue,
+                                        currencyData: currencyOptions,
+                                    });
+                                }}
+                                placeholder="Search"
+                                placeholderTextColor={themeColors.placeholderText}
+                            />
+                        </View>
+                        <View style={[styles.flex1]}>
+                            <SectionList
+                                bounces={false}
+                                indicatorStyle="white"
+                                keyboardShouldPersistTaps="always"
+                                showsVerticalScrollIndicator={false}
+                                sections={sections}
+                                keyExtractor={this.extractKey}
+                                stickySectionHeadersEnabled={false}
+                                renderItem={this.renderItem}
+                                renderSectionHeader={({section: {title}}) => (
+                                    <View>
+                                        <Text style={[styles.p5, styles.textMicroBold, styles.colorHeading]}>
+                                            {title}
+                                        </Text>
+                                    </View>
+                                )}
+                            />
+                        </View>
+                    </View>
+                    <View style={[styles.ph5, styles.pb5]}>
+                        <Pressable
+                            onPress={() => {
+                                Onyx.merge(ONYXKEYS.MY_PERSONAL_DETAILS, {
+                                    preferredCurrencyCode: this.state.selectedCurrency.currencyCode,
+                                    preferredCurrencySymbol: this.state.selectedCurrency.currencySymbol,
+                                });
+                                Navigation.goBack();
+                            }}
```

Comment:
Sorry maybe this is just personal preference 😅 but I like this in a class-scoped function too.